stakingModel/contracts/UBIScheme.sol

```diff
@@ -0,0 +1,268 @@
+            }
+            lastClaimed[account] = now; // marks last claimed as today
+            emit AddedToPending(account, lastClaimed[account]);
+        }
+        emit AlreadyClaimed(account, lastClaimed[account]);
+        return false;
+    }
+
+    /* @dev Function for claiming UBI. Requires contract to be active and claimer to be whitelisted.
+     * Calls distributionFormula, calculating the amount the caller can claim, and transfers the amount
+     * to the caller. Emits the address of caller and amount claimed.
+     * @return A bool indicating if UBI was claimed
+     */
+    function claim()
+        public
+        requireActive
+        onlyWhitelisted
+        returns (bool)
+    {
+        return _claim(msg.sender);
+    }
+
+    /* @dev In order to update users from active to inactive, we give out incentive to people
+     * to update the status of inactive users, this action is called "Fishing". Anyone can
+     * send a tx to the contract to mark inactive users. The "fisherman" receives a reward
+     * equal to the daily UBI (ie instead of the “fished” user). User that “last claimed” > 14
+     * can be "fished" and made inactive (reduces active users count by one). Requires
+     * contract to be active.
+     * @param account to fish
+     * @return A bool indicating if UBI was fished
+     */
+    function fish(address account)
+        public
+        requireActive
+        returns (bool)
+    {
+        setDay();
+        require(isRegistered(account) && !isActiveUser(account), "is not an inactive user");
+        require(!fishedUsersAddresses[account], "already fished");
+        fishedUsersAddresses[account] = true; // marking the account as fished so it won't refish
+
+        // making sure that the calculation will be with the correct number of active users in case
+        // that the fisher is the first to make the calculation today
+        uint256 newDistribution = distributionFormula(0, account);
+        activeUsersCount = activeUsersCount.sub(1);
+        GoodDollar token = GoodDollar(address(avatar.nativeToken()));
+        token.transfer(msg.sender, newDistribution);
+        Day memory day = claimDay[currentDay];
+        day.amountOfClaimers = day.amountOfClaimers.add(1);
+        day.claimAmount = day.claimAmount.add(newDistribution);
+        claimDay[currentDay] = day;
+        emit UBIFished(msg.sender, account, newDistribution);
+        return true;
+    }
+
+    /* @dev executes `fish` with multiple addresses
+     * @param accounts to fish
+     * @return A bool indicating if all the UBIs were fished
+     */
+    function fishingPool(address[] memory accounts)
+        public
+        requireActive
+        returns (bool)
+    {
+        require(accounts.length < gasleft().div(iterationGasLimit), "exceeds of gas limitations");
+        for(uint256 i = 0; i < accounts.length && i < activeUsersCount; ++i) {
+            fish(accounts[i]);
```

Comment:
also here activeUsersCount is irrelevant.
i'll open a separate issue to handle gas in loops

Reply:
updated
89bbc9109fbba53e9fc6320604bab47dcd85de87